src/query.rs

```diff
@@ -184,4 +208,612 @@ pub fn scanlog(text: &String, flags: ScanFlags) -> HashMap<String, Vec<String>>
+                                                positional_fields.push(PositionalColumn { position: position, alias: identifier.clone() });
+                                            } else {
+                                                // try to parse as as smart field
+                                                for sfield in SMART_FIELDS_RE.captures_iter(&identifier[..]) {
+                                                    let typed = sfield[2].to_string();
+                                                    let mut pos = 1;
+                                                    if sfield.get(4).is_none() == false {
+                                                        pos = match sfield[4].parse::<i32>() {
+                                                            Ok(p) => p,
+                                                            // technically this should never happen as the regex already validated an integer
+                                                            Err(_) => -1,
+                                                        };
+                                                    }
+                                                    // we use this set to keep track of active smart fields
+                                                    smart_fields_set.insert(typed.clone());
+                                                    // track the smartfield
+                                                    smart_fields.push(SmartColumn { typed: typed.clone(), position: pos, alias: identifier.clone() });
+                                                }
+                                            }
+                                        }
+                                        _ => {
+                                            info!("Unhandled operation");
+                                        }
+                                    }
+                                }
+                            }
+                            _ => {}
+                        }
+                    }
+                    _ => {}
+                };
+
+                // Build the parsing flags used by scanlog
+                let mut scan_flags: ScanFlags = ScanFlags::NONE;
+                for sfield_type in smart_fields_set {
+                    let flag = match sfield_type.as_ref() {
+                        "$ip" => ScanFlags::IP,
+                        "$email" => ScanFlags::EMAIL,
+                        "$date" => ScanFlags::DATE,
+                        "$quoted" => ScanFlags::QUOTED,
+                        "$url" => ScanFlags::URL,
+                        _ => ScanFlags::NONE,
+                    };
+                    if scan_flags == ScanFlags::NONE {
+                        scan_flags = flag;
+                    } else {
+                        scan_flags = scan_flags | flag;
+                    }
+                }
+//                println!("flags: {:?}", scan_flags);
```

Comment:
Do we need to keep this code?

Reply:
not really, It is useful for me hahaha, I'll remove it